scripts/veloxbench/veloxbench/cpp_micro_benchmarks.py

```diff
@@ -56,28 +57,45 @@ class LocalCppMicroBenchmarks:
     # see TODO about COMMON_OPTIONS above, set an empty dict for now
     options = copy.deepcopy(COMMON_OPTIONS)
     options.update(**RUN_OPTIONS)
-    description = "Run the Velox C++ micro benchmarks."
+    description = "Run Velox C++ micro-benchmarks."
     iterations = 1
     flags = {"language": "C++"}
 
-    def run(self, result_dir, **kwargs):
-        binaries = self._find_binaries(
-            REPO_ROOT.joinpath("_build", "release", "velox", "benchmarks", "basic")
-        )
+    def run(
+        self,
+        result_dir,
+        binary_path=None,
+        binary_filter=None,
+        bm_filter=None,
+        bm_max_secs=10,
+        bm_max_trials=1000000,
+        **kwargs,
+    ):
+        if binary_path:
+            binary_path = self._normalize_path(binary_path)
+        else:
+            binary_path = self._default_binary_path()
+
+        binaries = self._find_binaries(binary_path)
         result_dir_path = pathlib.Path(result_dir)
 
         for binary_path in binaries:
+            if binary_filter and not re.search(binary_filter, binary_path.name):
+                continue
+
             out_path = result_dir_path / f"{binary_path.name}.json"
             print(f"Executing and dumping results for '{binary_path}' to '{out_path}':")
             run_command = [
                 binary_path,
                 "--bm_max_secs",
-                "10",
+                str(bm_max_secs),
                 "--bm_max_trials",
```

Comment:
service lab has this flags
not sure what it does and how it make things more accuarate:
    "--bm_estimate_time",  # estimate running time to have more accurate result

Reply:
Yeah, I'll experiment with this one as well. For now just want to make this one a bit more flexible to we can experiment with different configurations.